action.yaml

```diff
@@ -0,0 +1,366 @@
+---
+name: Setup R dependencies
+author: Insights Engineering
+description: GitHub Action to install R package dependencies
+
+inputs:
+  github-token:
+    description: GitHub token.
+    required: false
+    default: ""
+  repository-path:
+    description: Directory where the R package is located relative to the calling GitHub workflow workspace.
+    required: false
+    default: "."
+  lookup-refs:
+    description: |
+      List of package references to be used for the feature branch.
+      Multiple entries in new lines or separated by commas.
+    required: false
+    default: ""
+  extra-deps:
+    description: |
+      List of extra dependencies to check against feature branch existence.
+      Multiple entries in new lines or separated by commas. It allows to control indirect dependencies.
+      This requires an entry in the `lookup-refs` input.
+    required: false
+    default: ""
+  dependencies:
+    description: Passed to `r-lib/actions/setup-r-dependencies`.
+    required: false
+    default: '"all"'
+  extra-packages:
+    description: Passed to `r-lib/actions/setup-r-dependencies`.
+    required: false
+    default: ""
+  needs:
+    description: |
+      Passed to `r-lib/actions/setup-r-dependencies`.
+      The value will be amended by `DepsBranch` and `DepsDev` values.
+    required: false
+    default: ""
+  skip-desc-branch:
+    description: Skip the DESCRIPTION file modification for the feature branch.
+    required: false
+    default: false
+  skip-desc-dev:
+    description: Skip the DESCRIPTION file modification for the development branch.
+    required: false
+    default: false
+  skip-install:
+    description: Skip the installation of R package dependencies.
+    required: false
+    default: false
+
+branding:
+  icon: 'download'
+  color: 'orange'
+
+runs:
+  using: composite
+  steps:
+
+  - name: Get branch names
+    id: branch-names
+    uses: tj-actions/branch-names@v8
+
+  - name: Install required packages
+    uses: r-lib/actions/setup-r-dependencies@v2
+    with:
+      packages: |
+        any::sessioninfo
+      extra-packages: |
+        any::pkgdepends
+        any::cli
+        any::desc
+      install-pandoc: false
+
+  - name: Prepare DESCRIPTION file (feature branch)
+    if: >
+      inputs.skip-desc-branch == 'false' ||
+      steps.branch-names.outputs.is_default == 'true'
+    id: desc-branch
+    shell: Rscript {0}
+    run: |
+      cat("::group::Define utils\n")
+      base_packages <- c("R", "base", "utils", "stats", "graphics", "grDevices", "datasets", "methods", "tools", "parallel", "compiler")
+      desc_field_create <- function(d, key) {
```

Comment:
Do you plan on moving the functions to an R file and `source()` it / `box::use()`?

I understand keeping the code in `yaml`, but duplicating definitions seems problematic.

Reply:
It's definitely an alternative. To be honest, I just wanted to try code in yaml. I found it quite handy for passing values from the outside (e.g. action parameters). This is also nice as the whole logic is in just one file - no need to switch between yaml and script.
It is true that this file is quite long and there are some repetitions. It might be due to the fact that I tried to write many functions so that the _main_ part of the code can be read nicely. If you look closely there are a functions with just one call. There are also one-liner functions. However, I quite like how it is right now (obviously I'm biased) just because of readability.
We can also move (some of) "Assertions" elsewhere - presumably: `if` clause of an action.
If you have any other ideas or you thing I went to far with readability - let me know.

Reply:
It would be awesome if we could create an environment and install it as a package to `.libPath` programmatically :-) 

To keep it all in yaml I can suggest saving the image after function definitions

```r
# ...
parse_input <- function(input) {
  gsub("\n", ",", input) |>
    strsplit(x = _, split = ",") |>
    _[[1]] |>
    trimws()
}
save.image("temporary_function.R")
```

And load it at the beginning of the next step. With a simple `load()` or into an environment.

```r
load("temporary_function.R", envir = tryCatch(as.environment("action_functions"), error = function(err) attach(NULL, name = "action_functions")))
```

Reply:
So there are two things here:
- write permission and I think this should be good
- temp file location. We can't store it in working directory where the package is because then R CMD BUILD will complain about non-standard file. (we can add to `.Rbuildignore` but I don't want to expand that much)

But as a general idea - I like this to move it before and then _load_ it somehow. Let me think about it more.

Reply:
OK this is already there. Thanks for a good idea 👍